win32ss/user/user32/windows/input.c

```diff
@@ -657,11 +657,18 @@ IntLoadKeyboardLayout(
     UNICODE_STRING ustrKLID;
     WCHAR wszRegKey[256] = L"SYSTEM\\CurrentControlSet\\Control\\Keyboard Layouts\\";
     WCHAR wszLayoutId[10], wszNewKLID[10], szImeFileName[80];
+    PWCHAR endptr;
     HKL hNewKL;
     HKEY hKey;
     BOOL bIsIME;
 
-    dwhkl = wcstoul(pwszKLID, NULL, 16);
+    dwhkl = wcstoul(pwszKLID, &endptr, 16);
+    if (endptr - pwszKLID != 8)
```

Comment:
So, if there was some whitespace before the first character in the string pointed by `pwszKLID`, or if there was a `0x` prefix, you would consider this to be invalid?

Reply:
Added a new function to validate, that uses `wcsspn` function.